### PR TITLE
c webcrawler settings UI

### PR DESCRIPTION
**Pull request by lasryaric**
## Description

UI side of the Webcrawler settings.

This is not following any Figma since @Duncid is working on a new version of this page, but it's part of a bigger work that is not ready yet.

Screenshot 2024-02-08 at 11 14 52

## Deploy Plan

- deploy front
- deploy connectors
- `npm run initdb` on front to drop the useless index on the  data_sources table.

### CODE REVIEW

File: front/pages/api/w/[wId]/data_sources/managed.ts [hunk 1]
```diff
@@ -27,11 +28,11 @@ import { apiError, withLogging } from "@app/logger/withlogging";
 
 const { NODE_ENV } = process.env;
 
-const PostManagedDataSourceRequestBodySchema = t.type({
+export const PostManagedDataSourceRequestBodySchema = t.type({
   provider: t.string,
   connectionId: t.union([t.string, t.undefined]),
   type: t.union([t.literal("oauth"), t.literal("url")]),
-  url: t.union([t.string, t.undefined]),
+  urlPayload: t.union([CreateConnectorUrlRequestBodySchema, t.undefined]),
```
Comment (flvndvd) [inline front/pages/api/w/[wId]/data_sources/managed.ts:35]: Let's align with the other places and go with `urlConfig` WDYT?

File: front/pages/w/[wId]/builder/data-sources/new-public-url.tsx [hunk 2]
```diff
@@ -153,39 +165,135 @@ export default function DataSourceNew({
...
-                    "block w-full min-w-0 flex-1 rounded-md  text-sm",
-                    dataSourceNameError
-                      ? "border-gray-300 border-red-500 focus:border-red-500 focus:ring-red-500"
-                      : "border-gray-300 focus:border-action-500 focus:ring-action-500"
-                  )}
+      <div className="flex flex-col space-y-8 pb-16 pt-8">
+        <div className="flex w-full flex-col gap-4">
+          <div className="flex flex-row items-start gap-8">
+            <div className="flex flex-col gap-4">
+              <div className="text-lg font-bold text-element-900">
```
Comment (flvndvd) [inline front/pages/w/[wId]/builder/data-sources/new-public-url.tsx:172]: It makes more sense to use `<p>` just to surround text elements.

File: front/pages/w/[wId]/builder/data-sources/new-public-url.tsx [hunk 3]
```diff
@@ -114,12 +122,16 @@ export default function DataSourceNew({
...
+          url: dataSourceUrl,
+          maxPages: maxPages || WEBCRAWLER_MAX_PAGES,
+          depth: maxDepth || WEBCRAWLER_MAX_DEPTH,
+          crawlMode: crawlMode,
+        },
         type: "url",
         provider: "webcrawler",
-      }),
+        connectionId: undefined,
+      } satisfies t.TypeOf<typeof PostManagedDataSourceRequestBodySchema>),
```
[the remark this answers is not in the document]
Comment (lasryaric) [reply front/pages/w/[wId]/builder/data-sources/new-public-url.tsx:134]: I don't get it? Is that for the satisfies?

File: front/pages/w/[wId]/builder/data-sources/new-public-url.tsx [hunk 4]
```diff
@@ -153,39 +165,131 @@ export default function DataSourceNew({
...
+                  }}
+                  name="crawlMode"
+                  choices={[
+                    {
+                      label: "Only sub pages.",
+                      value: "child",
+                      disabled: false,
+                    },
+                    {
+                      label: "The entier website.",
```
Comment (spolu) [inline front/pages/w/[wId]/builder/data-sources/new-public-url.tsx:214]: Entire ?

File: front/pages/w/[wId]/builder/data-sources/new-public-url.tsx [hunk 5]
```diff
@@ -61,6 +66,9 @@ export default function DataSourceNew({
 
   const [dataSourceNameError, setDataSourceNameError] = useState("");
   const [dataSourceUrl, setDataSourceUrl] = useState("");
+  const [maxPages, setMaxPages] = useState<number | null>(WEBCRAWLER_MAX_PAGES);
+  const [maxDepth, setMaxDepth] = useState<number | null>(WEBCRAWLER_MAX_DEPTH);
```
Comment (spolu) [inline front/pages/w/[wId]/builder/data-sources/new-public-url.tsx:70]: Same here, maybe 2 ?

File: front/pages/w/[wId]/builder/data-sources/new-public-url.tsx [hunk 6]
```diff
@@ -153,39 +165,131 @@ export default function DataSourceNew({
...
+                  name="dataSourceUrl"
+                  showErrorLabel
+                  className="text-sm"
                 />
               </div>
-              <p className="mt-2 text-sm text-gray-500">
-                This is the highest level URL on the selected domain that will
-                be crawled.
-              </p>
+              <div className="text-lg font-bold text-element-900">
```
Comment (flvndvd) [inline front/pages/w/[wId]/builder/data-sources/new-public-url.tsx:192]: `nits` All of those `<div>` could be regular `<p>`.

**Review by flvndvd — APPROVED** [review]

**Pull request merged**